### ISSUE TITLE
Cannot parse constructor references of array types when the element type is parameterized with "?"

### ISSUE THREAD

**Issue by miyakawataku** [checkstyle/checkstyle#1253]
Checkstyle 6.8-SNAPSHOT (ec414646eaf9ceb7de4b6e75b6c11d446cd0971) cannot parse constructor references of array types when the element type is parameterized with "?".

Example source code:

```java
import java.util.ArrayList;
import java.util.function.IntFunction;

public class ArrayConstructorReference {
    public static void main(String[] args) {
        IntFunction<?> fun = ArrayList<?>[]::new;
    }
}
```

The source code is valid and accepted by javac, but Checkstyle outputs an error message as follows:

```
/home/taku/pj/array/ArrayConstructorReference.java:6:39: expecting SEMI, found '<'
Jun 28, 2015 9:44:46 PM com.puppycrawl.tools.checkstyle.TreeWalker processFiltered
SEVERE: MismatchedTokenException occurred during the analysis of file ArrayConstructorReference.java.
/home/taku/pj/array/ArrayConstructorReference.java:7:5: expecting EOF, found '}'
        at antlr.Parser.match(Parser.java:211)
        at com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaRecognizer.compilationUnit(GeneratedJavaRecognizer.java:210)
        at com.puppycrawl.tools.checkstyle.TreeWalker.parse(TreeWalker.java:456)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:191)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:79)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:260)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:249)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:90)
```

**Comment by sabaka** [comment]
@romani ,
Looks like same problem with https://github.com/checkstyle/checkstyle/issues/2729
Reason is "[]", not '?'. It can be parsed without [].

**Comment by romani** [comment]
closed in favour of #2729
